src/fondant/core/schema.py

```diff
@@ -161,11 +162,34 @@ def __eq__(self, other):
         return False
 
 
-class Field(t.NamedTuple):
-    """Class representing a single field or column in a Fondant subset."""
+@dataclass
+class Field:
+    """Class representing a single field or column in a Fondant dataset."""
 
-    name: str
-    type: Type
+    def __init__(
+        self,
+        name: str,
+        type: Type = None,
+        location: str = "",
+    ) -> None:
+        self._name = name
+        self._type = type
+        self._location = location
+
+    @property
+    def name(self) -> str:
+        """The name of the field."""
+        return self._name
+
+    @property
+    def type(self) -> Type:
+        """The absolute location of the field."""
+        return self._type
+
+    @property
+    def location(self) -> str:
+        """The absolute location of the field."""
```

Comment:
```suggestion
        """The relative location of the field."""
```
You seem to be using this as a relative location in the other classes.